src/rf/apps/uploads/views.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.contrib.auth.models import User
+
+from rest_framework import serializers, viewsets
+from rest_framework.response import Response
+from rest_framework import decorators
+from rest_framework.permissions import AllowAny
+from rest_framework.renderers import StaticHTMLRenderer
+
+from django.conf import settings
+from django.shortcuts import render_to_response
+
+import base64
+import hmac, sha
+
+
+@decorators.api_view(['GET'])
+@decorators.permission_classes((AllowAny, ))
+@decorators.renderer_classes((StaticHTMLRenderer, ))
+def sign_upload_request(request, format=None):
```

Comment:
I think that we should import Boto for this and try to reuse its functions to derive the signature. Currently, S3 supports several signature versions, but V4 (which I don't think this code is compatible with) is the most recent. Using Boto will automatically keep us up-to-date.